internal/services/servicenetworking/application_load_balancer_association_resource.go

```diff
@@ -0,0 +1,227 @@
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(t.ResourceType(), id)
+			}
+
+			controller, err := trafficControllerClient.Get(ctx, *albId)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *albId, err)
+			}
+
+			association := associationsinterface.Association{
+				Properties: &associationsinterface.AssociationProperties{
+					Subnet: &associationsinterface.AssociationSubnet{
+						Id: config.SubnetId,
+					},
+					AssociationType: associationsinterface.AssociationTypeSubnets,
+				},
+			}
+
+			if controller.Model != nil {
+				association.Location = location.Normalize(controller.Model.Location)
+			}
+
+			if len(config.Tags) > 0 {
+				association.Tags = &config.Tags
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, association); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (t AssociationResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ServiceNetworking.AssociationsInterface
+
+			id, err := associationsinterface.ParseAssociationID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retreiving %s: %v", id.ID(), err)
+			}
+
+			trafficControllerId := associationsinterface.NewTrafficControllerID(id.SubscriptionId, id.ResourceGroupName, id.TrafficControllerName)
+			state := AssociationModel{
+				Name:                      id.AssociationName,
+				ApplicationLoadBalancerId: trafficControllerId.ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				state.Tags = pointer.From(model.Tags)
+
+				if prop := model.Properties; prop != nil {
+					if prop.Subnet != nil {
+						parsedSubnetId, err := commonids.ParseSubnetID(prop.Subnet.Id)
+						if err != nil {
+							return err
+						}
+						state.SubnetId = parsedSubnetId.ID()
+					}
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (t AssociationResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ServiceNetworking.AssociationsInterface
+
+			var plan AssociationModel
+			if err := metadata.Decode(&plan); err != nil {
+				return fmt.Errorf("decoding %v", err)
+			}
+
+			id, err := associationsinterface.ParseAssociationID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing id %v", err)
+			}
+
+			// Thought `AssociationSubnetUpdate` defined in the SDK contains the `subnetId`, while per testing it can not be updated
```

Comment:
@ziyeqf - have you opened an issue yet? could you add it to this comment? and address the question?

Reply:
updated the issue link to the code comment.
And yes for `CreateOrUpdate` it will get `AppGwForContainersSubnetAssociationChangeOfSubnetNotAllowed` error